### PR TITLE
[16.0][MIG] purchase_deposit: Migration to 16.0

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
Supersede of https://github.com/OCA/purchase-workflow/pull/1751.
@qrtl

### CODE REVIEW

File: purchase_deposit/wizard/purchase_make_invoice_advance.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        comodel_name="account.tax",
+        string="Vendor Taxes",
+        compute="_compute_deposit_account",
+        store=True,
+        readonly=False,
+        help="Taxes used for deposits",
+    )
+
+    @api.depends("purchase_deposit_product_id")
+    def _compute_deposit_account(self):
```
Comment (yostashiro) [inline purchase_deposit/wizard/purchase_make_invoice_advance.py]: Just to follow the convention.

```suggestion
    def _compute_deposit_account_id(self):
```

File: purchase_deposit/tests/test_purchase_deposit.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        self.po.button_confirm()
+        with Form(CreateDeposit.with_context(**ctx)) as f:
+            f.advance_payment_method = "percentage"
+        wizard = f.save()
+        wizard.amount = 10.0  # 10%
+        wizard.deposit_account_id = self.account_deposit
+        wizard.create_invoices()
+        # New Purchase Deposit is created automatically
+        deposit_id = self.env.company.purchase_deposit_product_id.id
+        deposit = self.product_model.browse(deposit_id)
```
Comment (yostashiro) [inline purchase_deposit/tests/test_purchase_deposit.py]: Same for other parts in this file where the same lines are repeated.

```suggestion
        deposit = self.env.company.purchase_deposit_product_id
```

File: purchase_deposit/tests/test_purchase_deposit.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        ctx = {
+            "active_id": self.po.id,
+            "active_ids": [self.po.id],
+            "active_model": "purchase.order",
+            "create_bills": True,
+        }
+        CreateDeposit = self.env["purchase.advance.payment.inv"]
+        self.po.button_confirm()
+        with Form(CreateDeposit.with_context(**ctx)) as f:
+            f.advance_payment_method = "percentage"
```
Comment (yostashiro) [inline purchase_deposit/tests/test_purchase_deposit.py]: Can this part be made into a separate method to be called from each test?

**Review by yostashiro — APPROVED** [review]
Code review and functional test.

**Review by florenciafrigieri2 — APPROVED** [review]
Functional test. Working ok!

**Comment by yostashiro** [comment]
@AungKoKoLin1997 Can you squash the last two commits please.

File: purchase_deposit/models/account_move.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def action_post(self):
+        res = super().action_post()
+        deposit_lines = self.line_ids.filtered(lambda x: x.purchase_line_id.is_deposit)
+        for line in deposit_lines:
```
Comment (RodrigoBM) [inline purchase_deposit/models/account_move.py]: why run two for when you can do it in one for?
```suggestion
        for line in self.line_ids:
            if not line.purchase_line_id.is_deposit:
                continue
      
```

File: purchase_deposit/models/purchase.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def copy_data(self, default=None):
+        if default is None:
+            default = {}
+        default["order_line"] = [
+            (0, 0, line.copy_data()[0])
+            for line in self.order_line.filtered(lambda l: not l.is_deposit)
```
Comment (RodrigoBM) [inline purchase_deposit/models/purchase.py]: why run two for when you can do it in one for?

```suggestion
            for line in self.order_line if not line.is_deposit
```

**Review by RodrigoBM — CHANGES_REQUESTED** [review]

File: purchase_deposit/models/account_move.py (hunk 4 above)
Comment (AungKoKoLin1997) [reply purchase_deposit/models/account_move.py]: I think adding one extra line for deposit_lines is better than iteration for unnecessary lines and adding condition inside loop.

Comment (AungKoKoLin1997) [reply purchase_deposit/models/account_move.py]: Sorry, you right. I lacked this knowledge https://github.com/OCA/purchase-workflow/pull/904#discussion_r419240295.

**Comment by rinaldifirdaus** [comment]
Functional Test: LGTM!

**Review by amarcosg — COMMENTED** [review]
Changes looks good, can you squash the commits? Thanks

**Comment by SoniaViciana** [comment]
**Functional review**: No ok

image


In which cases should this button be used? I think it can be removed.

Thanks,

**Comment by AungKoKoLin1997** [comment]
@SoniaViciana Thanks for the review. "CREATE DEPOSIT" button only creates deposit line in PO that was changed in  this commit https://github.com/OCA/purchase-workflow/commit/6f799e4367eb5bff822f8d19002473000ac07f22. I don't see the background reason of this changes. As far as I know, there is no reason to add only deposit line without create bill. I replace with "CREAET BILL" button and will create deposit bill but will not redirect to bill view. If someone ping me the background reason, I will revert the changes.

**Review by RodrigoBM — APPROVED** [review]

**Review by amarcosg — APPROVED** [review]
LGTM code review

**Comment by RodrigoBM** [comment]
can @HviorForgeFlow merge please?

**Pull request merged**